sections/appendix/problems.tex

```diff
@@ -0,0 +1,39 @@
+\subsection{Encountered Problems}
+
+\subsubsection{Too many low values}
+The document-topic distributions seem to have too low entropy and a lot of very low values.
+This causes each topic to have an average of $\sim$7000 docs in their distribution, however with very low entropy, meaning that some of these distributions are very insignificant, but still add space and time complexity to our sparse representations.
+To fix this we try tuning a minimum threshold for a topic to be counted in the document-topic distributions, which can be set as a hyper-parameter for \emph{gensim.ldamodel.get\_document\_topics()}.
+After setting up a minimum percent of 2.5\% for an entry to be included in the document distribution, the number of documents per topic fell to ~4800, removing more than 5 entries per document for a total of $\sim$350.000 entries deleted. 
+This has overall made the matrices a lot more sparse, but we will still have to evaluate whether changing K is better.
+
+\subsubsection{Empty Topics}
+There are also a significant number of topics, which had words, but no assigned docs. 
+These should be detected and removed, though their existence points towards a lower $K$ might be needed.
+
+\subsubsection{Too Common Topics}
+On the opposite end, we also have some topics which are so universal that almost all documents are included in this topic. 
+This might be a sign that pre-processing is not working well enough, or it might once again point at the needed for a better K.
+Alternatively, too general topics could be manually pruned after the LDA model has been run.
+
+\subsubsection{Adjacency Matrix Construction}
+Our adjacency matrix is based on whether documents share some topics in their topic distributions. 
+This leads us to the problem of D\*D\*T insertion operations into a sparse matrix with what was originally ~50.000 documents. (unknown)
+We did a lot of prepossessing, which reduced this number eventually all the way down to ~30.000 documents, but it was too insignificant to solve anything. (~140 days)
+However our similarity function is symmetric, so we cut the time in half by only constructing half of the adjacency matrix. (~69 days)
+We then changed our algorithm to only consider documents that shared topics to the original documents, decreasing our time to D\*T\_d\*D\_t, where both of the new variables are reduced compared to the originals. (~48 days)
+We also made the calculation multithreaded in order to reduce compution time (~2,4 days)
+Lastly, we changed to construct our matrix using the \emph{lil\_matrix} format, which has faster insertions.
+
+
+The function get\_document\_topics might be needing the whole corpus since whenever we give it the query it returns the same two topics. 
```

Comment:
Hvorfor nævner vi `get\_document\_topics`? Er det en Gensim funktion eller vores egen? Det er ikke tydeligt hvad det har med adjacency matrices at gøre.

Reply:
tror måske dette er meningen det skal være et seperat problem. Skulle ikke mene det er mig som har skrevet dette.

Reply:
```suggestion
\todo[inline]{Nedenstående problem skal uddybes, hvis det beholdes. Skal også have sit eget afsnit separat fra Adjacency Matrix Construction}
The function get\_document\_topics might be needing the whole corpus since whenever we give it the query it returns the same two topics. 
```